app/src/main/java/sdp/moneyrun/menu/JoinGameImplementation.java

```diff
@@ -176,10 +181,28 @@ private void createJoinButton(Button button, int buttonId, GameRepresentation ga
         button.setOnClickListener(v -> joinLobbyFromJoinButton(gameRepresentation));
 
         // Modify button if the game is full
-        if (gameRepresentation.getPlayerCount() >= gameRepresentation.getMaxPlayerCount()) {
-            button.setEnabled(false);
-            button.setText(activity.getString(R.string.join_game_full_message));
-        }
+        databaseReference.child(activity.getString(R.string.database_game))
+                .child(gameRepresentation.getGameId()).addValueEventListener(new ValueEventListener() {
+            @Override
+            public void onDataChange(@NonNull DataSnapshot snapshot) {
+                int newPlayerCount = (int) snapshot.child(activity.getString(R.string.database_open_games_players)).getChildrenCount();
+                if(newPlayerCount >= gameRepresentation.getMaxPlayerCount()){
+                    button.setEnabled(false);
+                    button.setText("Full");
```

Comment:
If this is the final text (not a dummy) that will be shown it should be saved as a resource.

Reply:
good point